src/cargo/ops/cargo_test.rs

```diff
@@ -100,11 +101,20 @@ fn run_unit_tests(options: &TestOptions,
             shell.status("Running", cmd.to_string())
         })?;
 
-        if let Err(e) = cmd.exec() {
-            errors.push(e);
-            if !options.no_fail_fast {
-                return Ok((Test::UnitTest(kind.clone(), test.clone()), errors))
+        let result = cmd.exec();
+
+        match result {
+            Err(CargoError(CargoErrorKind::ProcessErrorKind(e), .. )) => {
```

Comment:
Upon review, we need to move the `ProcessError` out here, but `kind` gives a borrow.